src/main/java/net/dv8tion/jda/internal/entities/ActivityImpl.java

```diff
@@ -95,7 +95,7 @@ public RichPresence.Timestamps getTimestamps()
 
     @Nullable
     @Override
-    public Emoji getEmoji()
+    public EmojiUnion getEmoji()
     {
         return emoji;
```

Comment:
This is only place that would need the cast if we removed the union requirement on the constructor.

Reply:
We only ever create the `ActivityImpl` using `Emoji.fromData` which already returns `EmojiUnion`, adding any casts at all to this is unnecessary.